src/sqlfluff/core/parser/segments/base.py

```diff
@@ -971,6 +971,11 @@ def iter_patches(self, templated_str: str) -> Iterator[FixPatch]:
             return
 
         # If we're here, the segment doesn't match the original.
+        linter_logger.debug("%s at %s: Original: [%r] Fixed: [%r]",
```

Comment:
This new message helps see how generating patches descends the parse tree to zero in on which segment(s) were changed.